### PR TITLE
[CSL-1831] Add generic result click tracking

### PR DESCRIPTION
**Pull request by stanlp1**

### CODE REVIEW

File: library/src/main/java/io/constructor/core/ConstructorIo.kt [hunk 1]
```diff
@@ -1123,6 +1124,48 @@ object ConstructorIo {
...
+     * ```
+     * ConstructorIo.trackGenericResultClick("Pencil", "123", "234")
+     * ```
+     * @param itemName the name of the item clicked
+     * @param customerId the id of the item clicked
+     * @param variationId the variationId of the item clicked
+     * @param sectionName section of the item clicked
+     */
+    fun trackGenericResultClick(itemName: String, customerId: String, variationId: String? = null, sectionName: String? = null) {
+        var completable = trackItemDetailLoadedInternal(itemName, customerId, variationId, sectionName)
```
Comment (jjl014) [inline library/src/main/java/io/constructor/core/ConstructorIo.kt:1139]: Should this be `trackGenericResultClickInternal` instead?

**Review by jjl014 — APPROVED** [review]
Nice, this is looking great to me!

Just left one comment around the `trackGenericResultClick` function

File: library/src/main/java/io/constructor/core/ConstructorIo.kt (hunk 1 above)
Comment (stanlp1) [reply library/src/main/java/io/constructor/core/ConstructorIo.kt]: Correct 👍

**Review by jjl014 — APPROVED** [review]
Awesome, this looks great to me!

I forgot to mention this previously, but let's also update the README with the new tracking call and add some info about when it should be used.

**Pull request merged**